src/wrapper_array.jl

```diff
@@ -166,14 +166,22 @@ for f in (:getindex, :view, :dotview)
         end
 
         @propagate_inbounds function Base.$f(a::NamedDimsArray, inds...)
-            # Some nonscalar case, will return an array, so need to give that names.
+            # Some nonscalar case, will return an array, probably with names
             data = Base.$f(parent(a), inds...)
             L = remaining_dimnames_from_indexing(dimnames(a), inds)
-            return NamedDimsArray{L}(data)
+            if L === nothing
+                # All names dropped, e.g. from mat[mat .> 0]
+                return data
+            else
+                return NamedDimsArray{L}(data)
+            end
         end
     end
 end
 
+remaining_dimnames_from_indexing(dimnames::Tuple{Any, Any, Vararg},
+    inds::Tuple{NamedDimsArray{L,T,N,<:BitArray} where {L,T,N}}) = nothing
```

Comment:
By BlueStyle this is not an acceptable way to do a line break
https://github.com/invenia/BlueStyle#method-definitions